app/daemon.go

```diff
@@ -312,57 +312,49 @@ func (d *MenderShellDaemon) dbusEventLoop(client mender.AuthClient) {
 			break
 		}
 
-		if d.needsReconnect() {
```

Comment:
In this newest commit I removed this extra check "if d.needsReconnect" early in the function which set the needsReconnect value to True. Rather the code now only have one check. Having these two checks doing the same thing must have delayed the tests and made it fail.